libfwupd/fwupd-client-sync.c

```diff
@@ -2676,3 +2676,62 @@ fwupd_client_emulation_save(FwupdClient *self, GCancellable *cancellable, GError
 	}
 	return g_steal_pointer(&helper->bytes);
 }
+
+
+static void
+fwupd_client_repair_cb(GObject *source, GAsyncResult *res, gpointer user_data)
+{
+	FwupdClientHelper *helper = (FwupdClientHelper *)user_data;
+	helper->ret =
+	    fwupd_client_repair_finish(FWUPD_CLIENT(source), res, &helper->error);
+	g_main_loop_quit(helper->loop);
+}
+
+/**
+ * fwupd_client_repair:
+ * @self: a #FwupdClient
+ * @repair_item: the AppStream_id
+ * @action: specify "do" or "undo"
```

Comment:
What does `undo` mean in this context?  Some of the things that could be "repaired" for HSI can't be undone from the OS.  For example turning on secure boot is a one way street.